tests/unit/s2n_cert_request_server_test.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <s2n.h>
+
+#include "tls/s2n_config.h"
+#include "tls/s2n_tls.h"
+#include "stuffer/s2n_stuffer.h"
+#include "s2n_test.h"
+
+typedef enum {
+    S2N_CERT_TYPE_RSA_SIGN = 1,
+    S2N_CERT_TYPE_DSS_SIGN = 2,
+    S2N_CERT_TYPE_RSA_FIXED_DH = 3,
+    S2N_CERT_TYPE_DSS_FIXED_DH = 4,
+    S2N_CERT_TYPE_RSA_EPHEMERAL_DH_RESERVED = 5,
+    S2N_CERT_TYPE_DSS_EPHEMERAL_DH_RESERVED = 6,
+    S2N_CERT_TYPE_FORTEZZA_DMS_RESERVED = 20,
+    S2N_CERT_TYPE_ECDSA_SIGN = 64,
+    S2N_CERT_TYPE_RSA_FIXED_ECDH = 65,
+    S2N_CERT_TYPE_ECDSA_FIXED_ECDH = 66,
+} s2n_cert_type;
+
+static uint8_t s2n_cert_type_preference_list[] = {
+        S2N_CERT_TYPE_RSA_SIGN,
+        S2N_CERT_TYPE_ECDSA_SIGN
+};
+
+static uint8_t s2n_cert_type_preference_list_legacy_dss[] = {
+        S2N_CERT_TYPE_RSA_SIGN,
+        S2N_CERT_TYPE_DSS_SIGN,
+        S2N_CERT_TYPE_ECDSA_SIGN
+};
+
+int main(int argc, char **argv)
+{
+
+    BEGIN_TEST();
+
+    {
```

Comment:
Could you add a comment explaining what this test is intending to do? It looks like it is supposed to test what the default behavior is of the cert request.

Reply:
Added!